### PR TITLE
Server: Email verification tokens and validators

### PR DESCRIPTION
**Pull request by promaty**

### CODE REVIEW

File: packages/server/src/errors/error-handler.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { DBError } from 'objection'
+import HttpStatus from 'http-status-codes'
+import { Errors as PostmarkErrors } from 'postmark'
```
[the remark this answers is not in the document]
Comment (promaty) [reply packages/server/src/errors/error-handler.js:3]: No actually destructuring doesn't work for imports.

File: packages/server/src/validators/UsersValidator.js [hunk 2]
```diff
@@ -1,25 +1,52 @@
...
+    if (!this.errors.length) {
+      await this._validateEmailChange({address, email})
+    }
+    return this.resetErrors()
+  }
+
+  async _validateAddressNew(address) {
+    const count = await User.getCount({address})
+    if (count > 0) {
+      return this.addError({address: 'Given address is already registered'})
```
[the remark this answers is not in the document]
Comment (promaty) [reply packages/server/src/validators/UsersValidator.js:32]: I know, was just being consistent.

File: packages/server/src/helpers/token-manager.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+function tokenGenerate(expiresIn='24h') {
+  const payload = { timestamp: Date.now() }
+  return jwt.sign(payload, EMAIL_JWT_PRIVATE_KEY, {expiresIn})
+}
+
+function tokenVerify(token) {
+  return jwt.verify(token, EMAIL_JWT_PRIVATE_KEY)
+}
+
+export { tokenGenerate, tokenVerify }
```
[the remark this answers is not in the document]
Comment (promaty) [reply packages/server/src/helpers/token-manager.js:16]: either is fine for me, I can change

**Comment by promaty** [comment]
Addressed all comments and created a shared email client in #154

File: packages/server/src/validators/BaseValidator.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import { isAddress } from 'web3-utils'
```
Comment (sohkai) [inline packages/server/src/validators/BaseValidator.js:1]: This comment is maybe coming too late, but I question the design of these `Validator`s and their abstraction mechanisms:

1. The state inside (`this.errors`) seems to be reset on every validation, leading me to believe we must be reusing the same validator instances?
1. `return resetErrors()`'s name and behaviour are opaque to readers who do not understand the base class's implementation

Perhaps it'd be simpler to just export one or more validation functions for each concern, or compose a `Validator` with a stateful `ValidationError` class?

Comment (promaty) [reply packages/server/src/validators/BaseValidator.js:1]: created issue #197

**Pull request merged**